hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/featurestore/featuregroup/FeaturegroupController.java

```diff
@@ -642,26 +619,26 @@ private void executeUpdateHiveQuery(String query, String databaseName, Project p
       if (stmt != null) {
         stmt.close();
       }
-      closeConnection(conn);
+      closeConnection(conn, user, project);
     }
     return resultList;
   }
-
+  
   /**
    * Checks if the JDBC connection to HS2 is open, and if so closes it.
+   *
+   * @param conn the JDBC connection
+   * @param user the user using the connection
+   * @param project the project where the connection is used
    */
-  private void closeConnection(Connection conn) {
+  private void closeConnection(Connection conn, Users user, Project project) {
     try {
       if (conn != null) {
         conn.close();
       }
-      //don't add this line:
-      //certificateMaterializer.removeCertificatesLocal(user.getUsername(), project.getName());
-      //concurrent requests by the same user will fail, let certs be cleaned up by
-      //garbage collector periodically instead
+      certificateMaterializer.removeCertificatesLocal(user.getUsername(), project.getName());
```

Comment:
`removeCertificatesLocal` in a finally block. `close` method can throw exceptions